### PR TITLE
Added onHeaderPress and onHeaderLongPress

### PR DESCRIPTION
**Pull request by bneigher**
This is added functionality which will not affect the UI at all without providing one of these functions to the Component props (Calendar, CalendarList, Agenda, ExpandableCalendar)

In short - if supplied, the View on the header gets replaced with a TouchableOpacity and the fn handler is invoked.  Can be used to or example hide or show the calendar, or change pages in your app.

### CODE REVIEW

**Comment by bneigher** [comment]
The feature is bullet proof guys - I promise!!
Hoping to get er merged so I can return to the wix distribution

**Comment by bneigher** [comment]
Screen Recording 2019-09-03 at 5 42 11 PM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,55 +1,84 @@
 # Changelog
 
+## ## [1.189.0] - 2019-07-16
```
Comment (Inbal-Tish) [inline CHANGELOG.md]: Please don't edit the CHANGELOG

File: .eslintrc.js [hunk 2]
```diff
@@ -1,44 +1,30 @@
 module.exports = {
-    "env": {
-        "es6": true,
-        "node": true
+  env: {
```
Comment (Inbal-Tish) [inline .eslintrc.js]: Why did you change this file? How it is relevant to your feature?

File: CHANGELOG.md (hunk 1 above)
Comment (Inbal-Tish) [reply CHANGELOG.md]: Regarding the README, I can see the changes so please remove them.

File: example/src/screens/calendars.js [hunk 3]
```diff
@@ -1,11 +1,6 @@
-import React, {Component} from 'react';
-import {
-  Text,
-  StyleSheet,
-  ScrollView,
-  View
-} from 'react-native';
-import {Calendar} from 'react-native-calendars';
+import React, { Component } from 'react';
```
Comment (Inbal-Tish) [inline example/src/screens/calendars.js]: Please remove all formatting changes.

**Comment by Inbal-Tish** [comment]
@bneigher Hi. Can you please explain the use case for this feature? Why would I need the header to be pressable? Can you provide a video to demonstrate? 
Also, there are conflicts in your PR and you should remove all formatting changes and changes to irrelevant files. Thanks!

File: .eslintrc.js (hunk 2 above)
Comment (bneigher) [reply .eslintrc.js]: I didn't change anything other than my IDE auto formats w.r.t. prettier style guide, so this file was simply autoindented.  It can be reverted back

**Comment by bneigher** [comment]
Here is the use case:  Essentially I want to be able to respond to a "close view" action, which in the case of my app makes a lot of sense to have attached to the calendar header bar.  Works great, and is optional ofcourse

ezgif-6-e0e707b05c76

File: example/src/screens/calendars.js (hunk 3 above)
Comment (bneigher) [reply example/src/screens/calendars.js]: done

File: CHANGELOG.md (hunk 1 above)
Comment (bneigher) [reply CHANGELOG.md]: done

File: example/src/screens/calendarsList.js [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import React, {Component} from 'react';
 import {CalendarList} from 'react-native-calendars';
 
-export default class CalendarList extends Component {
+export default class HorizontalCalendarList extends Component {
```
Comment (Inbal-Tish) [inline example/src/screens/calendarsList.js]: This is not the horizontal CalendarList example, why did you change the name?

**Comment by Inbal-Tish** [comment]
@bneigher Hi. Can you please remove all formating changes so I can review the actual feature's code?

File: example/src/screens/calendarsList.js (hunk 4 above)
Comment (bneigher) [reply example/src/screens/calendarsList.js]: my linter is complaining about the duplicate class names, but I'll disable it

**Comment by bneigher** [comment]
@Inbal-Tish - okay I did the best I could... to be honest there are several arbitrary fomatting characters like extra spaces after operators and inside of comments in the codebase, perhaps enforcing a auto-formatting standard to the project would be nice to prevent others from having this issue?

File: CHANGELOG.md [hunk 5]
```diff
@@ -284,4 +284,4 @@
 - Fix app icon.
 
 ## [1.261.0] - 2020-2-5
-- Calendar - adding accessibility features.
+- Calendar - adding accessibility features.
```
Comment (Inbal-Tish) [inline CHANGELOG.md]: Why there is a change here?

File: README.md [hunk 6]
```diff
@@ -503,4 +507,4 @@ See also the list of [contributors](https://github.com/wix/react-native-calendar
 
 Pull requests are most welcome! 
 Please `npm run test` and `npm run lint` before push.
-Don't forget to add a **title** and a **description** that explain the issue you're trying to solve and your suggested solution. Screenshots and gifs are very helpful.
+Don't forget to add a **title** and a **description** that explain the issue you're trying to solve and your suggested solution. Screenshots and gifs are very helpful.
```
Comment (Inbal-Tish) [inline README.md]: Why there is a change here?

File: example/src/screens/calendarsList.js [hunk 7]
```diff
@@ -1,12 +1,10 @@
 import React, {Component} from 'react';
 import {CalendarList} from 'react-native-calendars';
 
-
 export default class CalendarsList extends Component {
-  
   render() {
     return (
-      <CalendarList current={'2012-05-16'} pastScrollRange={24} futureScrollRange={24}/>
+      <CalendarList onHeaderPress={date => console.log('Header Pressed!', date)} current={'2012-05-16'} pastScrollRange={24} futureScrollRange={24}/>
```
Comment (Inbal-Tish) [inline example/src/screens/calendarsList.js]: please add an example for the `onHeaderLongPress` as well

File: src/calendar/header/index.js [hunk 8]
```diff
@@ -97,10 +101,17 @@ class CalendarHeader extends Component {
     return this.addMonth();
   }
 
-  render() {
+  onHeaderPress() {
+    this.props.onPress && this.props.onPress(this.props.month);
```
Comment (Inbal-Tish) [inline src/calendar/header/index.js]: You can safely invoke using lodash

File: src/calendar/header/index.js [hunk 9]
```diff
@@ -97,10 +101,17 @@ class CalendarHeader extends Component {
...
+  }
+
+  onHeaderLongPress() {
+    this.props.onLongPress && this.props.onLongPress(this.props.month);
```
Comment (Inbal-Tish) [inline src/calendar/header/index.js]: You can safely invoke using lodash

File: src/expandableCalendar/index.js [hunk 10]
```diff
@@ -60,7 +60,7 @@ class ExpandableCalendar extends Component {
     allowShadow: PropTypes.bool,
     /** whether to disable the week scroll in closed position */
     disableWeekScroll: PropTypes.bool
-  }
+  };
 
   static defaultProps = {
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: Please remove all formatting changes from this file

File: src/calendar/header/index.js [hunk 11]
```diff
@@ -145,6 +154,41 @@ class CalendarHeader extends Component {
 
     const webProps = Platform.OS === 'web' ? {'aria-level': this.props.webAriaLevel} : {};
 
+    if (this.props.onPress || this.props.onLongPress) {
```
Comment (Inbal-Tish) [inline src/calendar/header/index.js]: Why the code duplication? You only need to replace a View with TouchableOpacity when there is an onPress / onLongPress prop and pass a few more props...

File: src/calendar/index.js [hunk 12]
```diff
@@ -324,12 +352,15 @@ class Calendar extends Component {
           weekNumbers={this.props.showWeekNumbers}
           onPressArrowLeft={this.props.onPressArrowLeft}
           onPressArrowRight={this.props.onPressArrowRight}
+          onPress={this.props.onHeaderPress ? this.pressHeader : null}
```
Comment (Inbal-Tish) [inline src/calendar/index.js]: the prop check should be in the method, not here

File: src/calendar/index.js [hunk 13]
```diff
@@ -154,6 +174,14 @@ class Calendar extends Component {
     this._handleDayInteraction(date, this.props.onDayLongPress);
   }
 
+  pressHeader(date) {
+    this._handleHeaderInteraction(date, this.props.onHeaderPress);
```
Comment (Inbal-Tish) [inline src/calendar/index.js]: where the `pressHeader` gets the date parameter from?

File: src/calendar/index.js [hunk 14]
```diff
@@ -146,6 +151,21 @@ class Calendar extends Component {
     }
   }
 
+  _handleHeaderInteraction(date, interaction) {
```
Comment (Inbal-Tish) [inline src/calendar/index.js]: What is this method all about? Why is it necessary?

File: src/agenda/index.js [hunk 15]
```diff
@@ -44,6 +44,8 @@ export default class AgendaView extends Component {
     onDayPress: PropTypes.func,
     /** callback that gets called when day changes while scrolling agenda list */
     onDaychange: PropTypes.func,
+    /** callback that gets called on header press */
+    onHeaderPress: PropTypes.func,
```
Comment (Inbal-Tish) [inline src/agenda/index.js]: What about `onHeaderLongPress`? Where are you passing this prop to the CalendarList component?

File: example/src/screens/calendarsList.js (hunk 7 above)
Comment (Inbal-Tish) [reply example/src/screens/calendarsList.js]: I would actually move this example to the Calendars example screen so it will not appear as a CalendarList feature only

File: src/expandableCalendar/index.js [hunk 16]
```diff
@@ -69,22 +69,32 @@ class ExpandableCalendar extends Component {
...
-  }
+  };
 
   static positions = POSITIONS;
 
   constructor(props) {
     super(props);
 
     this.style = styleConstructor(props.theme);
-    this.closedHeight = CLOSED_HEIGHT + (props.hideKnob ? 0 : KNOB_CONTAINER_HEIGHT);
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: Please remove all formatting changes from this file

File: src/expandableCalendar/index.js [hunk 17]
```diff
@@ -331,41 +361,62 @@ class ExpandableCalendar extends Component {
...
+      // to allows setDate to be completed
       if (this.state.position === POSITIONS.OPEN) {
         this.bounceToPosition(this.closedHeight);
       }
     }, 0);
-  }
+  };
+
+  onHeaderPress = value => {
+    this.props.onHeaderPress && this.props.onHeaderPress(value);
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: You can invoke safely using lodash

File: src/expandableCalendar/index.js [hunk 18]
```diff
@@ -331,41 +361,62 @@ class ExpandableCalendar extends Component {
...
+  };
+
+  onHeaderLongPress = value => {
+    this.props.onHeaderLongPress && this.props.onHeaderLongPress(value);
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: You can invoke safely using lodash

File: src/expandableCalendar/index.js [hunk 19]
```diff
@@ -404,11 +464,30 @@ class ExpandableCalendar extends Component {
...
+            <Text allowFontScaling={false} style={this.style.headerTitle}>
+              {monthYear}
+            </Text>
+          </TouchableOpacity>
+        ) : (
+          <Text allowFontScaling={false} style={this.style.headerTitle}>
+            {monthYear}
+          </Text>
+        )}
+        <Text allowFontScaling={false} style={this.style.headerTitle}>
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: There's a duplication here

File: src/expandableCalendar/index.js [hunk 20]
```diff
@@ -404,11 +464,30 @@ class ExpandableCalendar extends Component {
...
-        pointerEvents={'none'}
-      >
-        <Text allowFontScaling={false} style={this.style.headerTitle}>{monthYear}</Text>
+        ref={e => (this.header = e)}
+        style={[
+          this.style.header,
+          {height: HEADER_HEIGHT, top: this.state.headerDeltaY}
+        ]}
+        pointerEvents={'none'}>
+        {this.props.onHeaderPress ? (
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: What about `onHeaderLongPress`?

File: src/expandableCalendar/index.js [hunk 21]
```diff
@@ -455,49 +533,66 @@ class ExpandableCalendar extends Component {
...
     return (
       <View style={[allowShadow && this.style.containerShadow, style]}>
-        {screenReaderEnabled ?
+        {screenReaderEnabled ? (
           <Calendar
             testID="calendar"
             {...others}
             theme={themeObject}
             onDayPress={this.onDayPress}
+            onHeaderPress={this.props.onHeaderPress && this.onHeaderPress}
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: You already have the prop check inside `this.onHeaderPress`

File: src/expandableCalendar/index.js [hunk 22]
```diff
@@ -455,49 +533,66 @@ class ExpandableCalendar extends Component {
...
+            onHeaderLongPress={
+              this.props.onHeaderLongPress && this.onHeaderLongPress
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: Same here

File: src/expandableCalendar/index.js [hunk 23]
```diff
@@ -455,49 +533,66 @@ class ExpandableCalendar extends Component {
...
+            }
+            onVisibleMonthsChange={this.onVisibleMonthsChange}
```
Comment (Inbal-Tish) [inline src/expandableCalendar/index.js]: Why did you add `onVisibleMonthChange` here? It's wrong. Please test this

**Comment by Inbal-Tish** [comment]
@bneigher Hi. I love your feature but there are some changes to make before I can merge it. Looking forward to your revised version. Thanks!

**Comment by bneigher** [comment]
@Inbal-Tish ahh this is a solid amount of work.. I probably can't get to this stuff any time soon unfortunately.  Was hoping ya'll could pick up where I left off.  I believe in general the required props to be able to handle the onPress and onLongPress are exposed (as I have this working on my app).

Unless you guys can pick up where I left off this may get stale.  That being said if anyone out there is interested in picking this up please go ahead!
